openstack/datadog_checks/openstack/openstack.py

```diff
@@ -1244,15 +1246,22 @@ def get_local_hypervisor(self):
         if hyp:
             return hyp[0]
 
-    def get_all_projects(self, scope):
+    def get_all_projects(self, scope, regexes):
         """
         Returns all projects in the domain
         """
+
         url = "{0}/{1}/{2}".format(self.keystone_server_url, DEFAULT_KEYSTONE_API_VERSION, "projects")
         headers = {'X-Auth-Token': scope.auth_token}
         try:
             r = self._make_request_with_auth_fallback(url, headers)
-            return r['projects']
+            filtered_projects = copy.deepcopy(r['projects'])
+            for project_match in r['projects']:
+               if regexes:
+                   for regex in regexes:
+                        if re.search(regex, project_match['name']):
+                            filtered_projects[:] = [project for project in filtered_projects if project.get('name') != project_match.get('name')]
```

Comment:
It would probably be more performant, and clearer if we started from an empty `filtered_projects`, and added only the ones that don't match any regex.
e.g. something like
```python
matches = re.search(regex, project_match["name"]) for regex in regexes
if not any(matches):
  filtered_projects.append(project_match)
```
What do you think ?